### PR TITLE
Feature: add new link for the "Add Form" admin menu to create forms with the new visual form builder

### PR DESCRIPTION
**Pull request by glaubersilva**
Resolves [GIVE-998]

## Description

This PR replaces the default _"Add Form"_ link [added by the WP Core](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/post-new.php#L36-L51) for custom post types with a new link that uses the visual form builder as default editor. 

As we don't have any filter to change this value, the `global $submenu;` var was used to accomplish that.

## Affects

The "Add Form" admin menu link.

## Visuals

Here is a sample of how the `global $submenu;` var looks like:

image

## Testing Instructions

1. Open the WP Admin area;
2. Click on the "Add Form" submenu from Give;
3. A new form using the Visual Form Builder should be created.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-998]: https://stellarwp.atlassian.net/browse/GIVE-998?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/DonationForms/DonationFormsAdminPage.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * @unreleased
+     *
+     * @param $giveSubmenu array [0: "Add Form", 1: "edit_give_forms", 2: "post-new.php?post_type=give_forms", ...]
+     */
+    private function isLegacyAddFormSubmenuLink(array $giveSubmenu): bool
+    {
+        return isset($giveSubmenu[2]) && 'post-new.php?post_type=give_forms' === $giveSubmenu[2];
+    }
```
Comment (pauloiankoski) [inline src/DonationForms/DonationFormsAdminPage.php:39]: WordPress provides tools to handle this beautifully without getting your hands dirty by changing the global variable directly. All you’ll need are these 2 lines of code:

```php
remove_submenu_page('edit.php?post_type=give_forms', 'post-new.php?post_type=give_forms');
add_submenu_page('edit.php?post_type=give_forms', 'Add Form', 'Add Form', 'edit_posts', 'edit.php?post_type=give_forms&page=givewp-form-builder');
```

**Review by pauloiankoski — CHANGES_REQUESTED** [review]

File: src/DonationForms/DonationFormsAdminPage.php (hunk 1 above)
Comment (glaubersilva) [reply src/DonationForms/DonationFormsAdminPage.php]: Oh man, much better! Thanks for the suggestion. Resolved in 04265c3aeb44e0bb9a45ebab8aee2b688470efd6

**Comment by glaubersilva** [comment]
@pauloiankoski This is ready for re-review.

**Review by pauloiankoski — APPROVED** [review]

**Review by rickalday — APPROVED** [review]
Pass manual QA tests.

**Pull request merged**